doc/stormond/stormond-hld.md

```diff
@@ -0,0 +1,263 @@
+
+```
+devices = {
+'mmcblk0' : None
+'sda'     : None
+}
+```
+
+**get_storage_device_object() Logic:**
+
+- For each key in the `devices` dictionary:
+    - If key starts with the term `sd`:
+        - If the realpath of `/sys/block/[KEY]/device` has the term `ata` in it:
+        - Instantiate an object<sup>READ NOTE</sup> of type `SsdUtil` and add this object as value of the key
+        ```
+        Example:
+        root@str-msn2700-02:~# cd /sys/block/sda/../../../0:0:0:0
+        root@str-msn2700-02:/sys/devices/pci0000:00/0000:00:1f.2/ata1/host0/target0:0:0/0:0:0:0#
+        ```
+    - else if the realpath of `/sys/block/[KEY]/device` has the term `usb` in it:
+        - Instantiate an object<sup>READ NOTE</sup> of type `UsbUtil` and add this object as value of the key
+        ```
+        Example:
+        root@str2-7050qx-32s-acs-01:~# cd /sys/block/sda/../../../2:0:0:0
+        root@str2-7050qx-32s-acs-01:/sys/devices/pci0000:00/0000:00:12.2/usb1/1-2/1-2:1.0/host2/target2:0:0/2:0:0:0#
+        ```
+- else if key starts with the term `mmcblk`:
+    - Instantiate an object<sup>READ NOTE</sup> of type `EmmcUtil` and add this object as value of the key
+    ```
+    Example:
+    root@str2-7050cx3-acs-01:/sys/block$ ls | grep -i "mmcblk" | grep -v "boot" | grep -v "loop"
+    mmcblk0
+    ```
+
+**Example usage:**
+
+Assuming a device contains the following storage disks:
+```
+root@str-a7280cr3-2:~# ls /sys/block/
+loop0  loop1  loop2  loop3  loop4  loop5  loop6  loop7  **mmcblk0**  mmcblk0boot0  mmcblk0boot1  **sda**
+```
+
+We would instantiate an object of the StorageDevices() class
+`storage = StorageDevices()`
+
+`storage.devices` would contain:
+```
+{
+    'mmcblk0': <Emmcutil object>,
+    'sda': <SsdUtil object>
+}
+```
+
+we would then get static and dynamic information by leveraging the respective member function implementations of `SsdUtil` and `EmmcUtil`, as they both derive from `SsdBase`.
+We then leverage the following proposed StateDB schema to store and stream information about each of these disks.
+
+
+<sub>**NOTE:** `UsbUtil` and `NVMeUtil` classes are not yet available. `EmmcUtil` class does not currently have IO reads, IO writes and Reserved Blocks support.</sub>
+
+## **3. StateDB Schema**
+```
+; Defines information for each Storage Disk in a device
+
+key                 = STORAGE_INFO|<disk_name>          ; This key is for information about a specific storage disk - STORAGE_INFO|SDX
+
+; field             = value
+
+temperature_celsius = STRING                            ; Describes the operating temperature of the SSD in Celsius                             (Priority 0, Dynamic)
+io_reads            = STRING                            ; Describes the total number of reads completed successfully from the SSD               (Priority 0, Dynamic)
+io_writes           = STRING                            ; Describes the total number of writes completed on the SSD                             (Priority 0, Dynamic)
```

Comment:
@assrinivasan please specify if these are filesytem read/write counts and what is the unit?

Reply:
Elaborated in latest commit